core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java

```diff
@@ -387,4 +393,55 @@ public void dataFilesCleanup() throws IOException {
     Assert.assertTrue("FILE_A should be deleted", deletedFiles.contains(FILE_A.path().toString()));
     Assert.assertTrue("FILE_B should be deleted", deletedFiles.contains(FILE_B.path().toString()));
   }
+  @Test
+  public void testWithExpiringWapThenCherrypick() {
+    // first WAP commit
+    table.newAppend()
+            .appendFile(FILE_B)
+            .stageOnly()
+            .commit();
+
+    TableMetadata base = readMetadata();
+    // pick the snapshot that's staged but not committed
+    Snapshot wap1Snapshot = base.snapshots().get(0);
```

Comment:
This shouldn't use the name `wap`, since this is independent of the WAP pattern.